server/src/main/java/org/elasticsearch/index/mapper/FieldMapper.java

```diff
@@ -1384,12 +1384,18 @@ public abstract static class Builder extends Mapper.Builder implements ToXConten
         protected Optional<SourceKeepMode> sourceKeepMode = Optional.empty();
         protected boolean hasScript = false;
         protected OnScriptError onScriptError = null;
+        protected final boolean withinMultiField;
 
         /**
          * Creates a new Builder with a field name
          */
         protected Builder(String name) {
+            this(name, false);
+        }
+
+        protected Builder(String name, boolean withinMultiField) {
             super(name);
+            this.withinMultiField = withinMultiField;
```

Comment:
Can we keep this in the `TextFieldMapper` and `MatchOnlyTextFieldMapper`? That duplicates the code but exposing `FieldMapper#isWithinMultiField` is dangerous IMO since none of the other builders are setting this correctly.